feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/balance/detail/BalanceDetailFragment.kt

```diff
@@ -110,26 +112,45 @@ class BalanceDetailFragment : BaseFragment<BalanceDetailViewModel>() {
         setupBuyIntegration(viewModel)
 
         viewModel.assetLiveData.observe { asset ->
-            balanceDetailTokenIcon.setImageResource(asset.token.configuration.icon)
-            balanceDetailTokenName.text = asset.token.configuration.name
+            balanceDetailTokenIcon.load(asset.token.configuration.iconUrl, imageLoader)
+            assetBadgeIcon.load(asset.token.configuration.iconUrl, imageLoader)
 
-            asset.token.dollarRate?.let {
-                balanceDetailDollarGroup.visibility = View.VISIBLE
+            balanceDetailTokenName.text = asset.token.configuration.symbol
+            balanceAssetName.text = asset.token.configuration.name
+
+            balanceDetailRate.text = asset.token.dollarRate?.formatAsCurrency() ?: ""
 
-                balanceDetailRate.text = it.formatAsCurrency()
-            }
 
             asset.token.recentRateChange?.let {
-                balanceDetailRateChange.setTextColorRes(asset.token.rateChangeColorRes!!)
+                balanceDetailRateChange.setTextColorRes(asset.token.rateChangeColorRes)
                 balanceDetailRateChange.text = it.formatAsChange()
             }
 
-            asset.dollarAmount?.let { balanceDetailDollarAmount.text = it.formatAsCurrency() }
+            balanceDetailsInfo.total.text = asset.total.formatTokenAmount(asset.token.configuration)
 
-            balanceDetailTotal.text = asset.total.formatTokenAmount(asset.token.configuration)
+            val totalFiat = asset.token.dollarRate?.let { asset.total.multiply(it).formatAsCurrency() }
+            if (totalFiat == null) {
+                balanceDetailsInfo.totalFiat.makeGone()
+            } else {
+                balanceDetailsInfo.totalFiat.text = totalFiat
+            }
 
-            balanceDetailFrozenAmount.text = asset.frozen.format()
-            balanceDetailAvailableAmount.text = asset.available.format()
+            balanceDetailsInfo.transferable.text = asset.available.formatTokenAmount(asset.token.configuration)
```

Comment:
maybe instead of copy-pasting code two times, I suppose this might show up somewhere on other screen/view also, let's make a fun that is like `TextView.applyDollarRateForAsset(asset: Asset)` and this logic is done within this function once